Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#include "Components/PSSpotComponent.h"
+
+#include "Components/PSHUDComponent.h"
+#include "Data/PSWorldSubsystem.h"
+#include "Data/PSSaveGameData.h"
+#include "Components/MySkeletalMeshComponent.h"
+#include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(PSSpotComponent)
+
+// Sets default values for this component's properties
+UPSSpotComponent::UPSSpotComponent()
+{
+	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
+	// off to improve performance if you don't need them.
+	PrimaryComponentTick.bCanEverTick = false;
+	PrimaryComponentTick.bStartWithTickEnabled = false;
+}
+
+// Called when the game starts
+void UPSSpotComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	PlayerSpotOnLevelInternal = GetMeshChecked();
+	checkf(PlayerSpotOnLevelInternal, TEXT("ERROR: 'PlayerSpotOnLevelInternal' is null"));
+
+	UPSWorldSubsystem::Get().OnCurrentRowDataChanged.AddDynamic(this, &ThisClass::OnPlayerTypeChanged);
+
+	PSHUDComponentInternal = UPSWorldSubsystem::Get().GetProgressionSystemHUDComponent();
+	checkf(PSHUDComponentInternal, TEXT("ERROR: 'PSHUDComponentInternal is null'"));
+
+	SaveGameInstanceInternal = UPSWorldSubsystem::Get().GetCurrentSaveGameData();
+	checkf(SaveGameInstanceInternal, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
+
+	ChangeSpotVisibilityStatus();
+}
+
+void UPSSpotComponent::OnUnregister()
```

Comment:
done https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/751d071c52eed4063782c4ee8d119a07e0276b25

Reply:
Not finished yet.

By 'cached', I refer specifically to `Transient` properties. Class Default properties (Non-`Transient`) like `PSDataAssetInternal` and widgets such as `PSCBackgroundOverlay` don't need resetting unless they are cached as `Transient` by another class. Resetting them is not mistake and will not bug anything, but it might confuse other programmers.

Please don't reset those.

Reply:
https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/252bfb763c62fe423fa1e9ee77c08288c09bcdb3 recovered